doc/fundamentals/append_overwrite.md

```diff
@@ -0,0 +1,616 @@
+```
+This would result in five datums being processed in any pipelines that use this repo.
+
+Now, take a one-line file containing
+```
+Six
+```
+and load it into Pachyderm appending it to the count.txt file.  If that file were named, "more-count.txt", the command might look like
+```
+pachctl put-file line-data master my-data.txt -f more-count.txt --split line
+```
+That will result in six files in the directory named "count.txt" in the input repo, each of which will have the following contents
+```
+count.txt/0000000000000000: One
+count.txt/0000000000000001: Two
+count.txt/0000000000000002: Three
+count.txt/0000000000000003: Four
+count.txt/0000000000000004: Five
+count.txt/0000000000000004: Six
+```
+This would result in one datum being processed in any pipelines that use this repo: the new file `count.txt/0000000000000004`.
+
+#### Overwriting files with --split
+The behavior of Pachyderm when a file loaded with `--split` is overwritten is simple to explain but subtle in its implications.  Remember that the loaded file will be split into those sequentially-named files, as shown above.  If any of those resulting split-files hashes differently than the one it's replacing, that will cause the Pachyderm Pipeline System to process that data.
+
+This can have important consequences for downstream processing.  For example, let's say you have that same file named "count.txt" consisting of 5 lines that we used in the previous example
+```
+One
+Two
+Three
+Four
+Five
+```
+As discussed prior, loading that file into Pachyderm using `--split` will result in five files in a directory named "count.txt" in the input repo, each of which will have the following contents
+```
+count.txt/0000000000000000: One
+count.txt/0000000000000001: Two
+count.txt/0000000000000002: Three
+count.txt/0000000000000003: Four
+count.txt/0000000000000004: Five
+```
+This would result in five datums being processed in any pipelines that use this repo.
+
+Now, modify that file by inserting the word "Zero" on the first line.
+```
+Zero
+One
+Two
+Three
+Four
+Five
+```
+Let's upload it to Pachyderm using `--split` and `--overwrite`.
+```
+pachctl put-file line-data master count.txt -f ./count.txt --split line --overwrite
+```
+The input repo will now look like this
+```
+count.txt/0000000000000000: Zero
+count.txt/0000000000000001: One
+count.txt/0000000000000002: Two
+count.txt/0000000000000003: Three
+count.txt/0000000000000004: Four
+count.txt/0000000000000005: Five
+``` 
+As far as Pachyderm is concerned, every single file existing has changed, and a new file has been added.  Six datums would be processed by a downstream pipelines.
+
+It's important to remember that what looks like a simple upsert can be a kind of a [fencepost error](https://en.wikipedia.org/wiki/Off-by-one_error#Fencepost_error).  Being "off by one line" in your data can be expensive, consuming processing resources you didn't intend to spend.
+
+### Datums in Pachyderm pipelines
```

Comment:
Upon reviewing this doc, I think the lifecycle of a datum and everything below here might make sense as a separate doc. Append.overwrite and split is one doc -- then we have the distributed computing doc with is like Datums 101 and this doc is like Datum 201.

Reply:
Split and added to this PR.